homeassistant/components/sht31/sensor.py

```diff
@@ -25,9 +33,40 @@
 DEFAULT_NAME = "SHT31"
 DEFAULT_I2C_ADDRESS = 0x44
 
-SENSOR_TEMPERATURE = "temperature"
-SENSOR_HUMIDITY = "humidity"
-SENSOR_TYPES = (SENSOR_TEMPERATURE, SENSOR_HUMIDITY)
+
+@dataclass
+class SHT31RequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[SHTClient], float | None]
+
+
+@dataclass
+class SHT31SensorEntityDescription(SensorEntityDescription, SHT31RequiredKeysMixin):
+    """Describes SHT31 sensor entity."""
+
+
+SENSOR_TYPES = (
+    SHT31SensorEntityDescription(
+        key="temperature",
+        name="Temperature",
+        device_class=DEVICE_CLASS_TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        value_fn=lambda sensor: sensor.temperature,
+    ),
+    SHT31SensorEntityDescription(
+        key="humidity",
+        name="Humidity",
+        device_class=DEVICE_CLASS_HUMIDITY,
+        native_unit_of_measurement=PERCENTAGE,
+        value_fn=lambda sensor: (
+            round(val)  # pylint: disable=undefined-variable
```

Comment:
Should we mention the pylint bug requiring this in the comment?

Reply:
Don't think that is necessary. I've already opened a PR to fix it. Should be included in the next release. https://github.com/PyCQA/pylint/pull/5044